logger.js

```diff
@@ -1,89 +1,75 @@
 dayjs.extend(require('dayjs/plugin/relativeTime'));
 
 /**
- * open database mode :
- * sqlite3.OPEN_CREATE | sqlite3.OPEN_READONLY | sqlite3.OPEN_READWRITE
+ * @class Logger
+ * @description this class allows to read and write logs in the database
  */
-const OPEN_MODE = 'sqlite3.OPEN_READWRITE';
-
 class Logger {
   constructor(name) {
-    this.db = new sqlite3.Database(name || './database.db', OPEN_MODE, (err) => {
-      if (err) { throw err; }
-      debug('Connected to the disk file database.');
-    });
+    this.db = new Database(name || './database.db');
   }
 
   write(params) {
     const start = dayjs(params.start).format('YYYY-MM-DD HH:mm:ss');
     const end = dayjs(params.end).format('YYYY-MM-DD HH:mm:ss');
     const elapsed = dayjs(end).from(start, true);
     const parameters = {
-      $googleDriveParentFolderId: params.googleDriveParentFolderId,
-      $normalizedStructure: params.normalizedStructure,
-      $normalizedReportName: params.normalizedReportName,
-      $attachments: params.attachments,
-      $start: start,
-      $sender: params.sender,
-      $end: end,
-      $elapsed: elapsed,
+      googleDriveParentFolderId: params.googleDriveParentFolderId,
+      normalizedStructure: params.normalizedStructure,
+      normalizedReportName: params.normalizedReportName,
+      attachments: params.attachments,
+      start,
+      sender: params.sender,
+      end,
+      elapsed,
     };
-    this.db.serialize(() => {
-      this.db.run(`
-        CREATE TABLE IF NOT EXISTS logger (
-          googleDriveParentFolderId INTEGER,
-          normalizedStructure TEXT,
-          normalizedReportName TEXT,
-          attachments TEXT,
-          start TEXT,
-          sender TEXT,
-          end TEXT,
-          elapsed TEXT
-        )`);
 
-      this.db.run(`
-        INSERT INTO logger (googleDriveParentFolderId, normalizedStructure, normalizedReportName, attachments, start, sender, end, elapsed)
-        VALUES ($googleDriveParentFolderId, $normalizedStructure, $normalizedReportName, $attachments, $start, $sender, $end, $elapsed)
-      `, parameters);
-    });
+    const createLoggerTableQuery = `
+      CREATE TABLE IF NOT EXISTS logger (
+        googleDriveParentFolderId INTEGER,
+        normalizedStructure TEXT,
+        normalizedReportName TEXT,
+        attachments TEXT,
+        start TEXT,
+        sender TEXT,
+        end TEXT,
+        elapsed TEXT
+      )`;
+
+    this.db
+      .prepare(createLoggerTableQuery)
+      .run();
+
+    const insertIntoLoggerQuery = `
+      INSERT INTO logger (googleDriveParentFolderId, normalizedStructure, normalizedReportName, attachments, start, sender, end, elapsed)
+      VALUES ($googleDriveParentFolderId, $normalizedStructure, $normalizedReportName, $attachments, $start, $sender, $end, $elapsed)
+    `;
+
+    this.db
+      .prepare(insertIntoLoggerQuery)
+      .run(parameters);
   }
 
   read(n) {
     return new Promise((resolve, reject) => {
```

Comment:
Since `better-sqlite3` is synchronous, let's get rid of the promise.  We can just directly run the code synchronously.  It should make this code a lot nicer to read. :)